### ISSUE TITLE
Method getGroupSortedChildNames when sort_order is not specified on block definition take the last only

### ISSUE THREAD

**Issue by magedevel** [magento/magento2#24474]
### Summary (*)

If we have a grouped block and we do not specify an order with argument sort_order, only the last block is rendered. 
If we specify the same  sort_order for grouped blocks only the last is rendered.
### Envirorment (*)
Magento CE 2.3.1
Nginx 
PHP 7.1
Valet +

### Examples (*)

Suppose we have in catalog_product_view.xml a definition like this:
```XML
<container name="bottom.content" htmlTag="div" htmlClass="bottom-content row" after="-">
                <container name="bottom.content.left" htmlTag="div" htmlId="bottom-content-left" htmlClass="col-md-8"before="-">
                    <block class="Magento\Catalog\Block\Product\View\Details" name="product.info.details"
                           template="Magento_Catalog::product/view/details.phtml" after="product.info.media">
                        <block class="Magento\Catalog\Block\Product\View\Attributes" name="product.info.description.new"
                               as="product.info.description.new"
                               template="Magento_Catalog::product/view/attributes.phtml" group="detailed_info" >
                            <arguments>
                                <argument name="at_call" xsi:type="string">getDescription</argument>
                                <argument name="at_code" xsi:type="string">description</argument>
                                <argument name="css_class" xsi:type="string">overview</argument>
                                <argument name="at_label" xsi:type="string">none</argument>
                                <argument name="title" translate="true" xsi:type="string">Details</argument>
                                <argument name="add_attribute" xsi:type="string">itemprop="description"</argument>

                            </arguments>
                        </block>
                        <block class="Magento\Catalog\Block\Product\View\Attributes"
                               name="product.info.custom.attribute" as="product.info.custom.attribute"
                               template="Magento_Catalog::product/view/custom-tab.phtml" group="detailed_info">
                            <arguments>
                                <argument translate="true" name="title" xsi:type="string">More Information</argument>
                            </arguments>
                        </block>
                    </block>
                </container>
                <container name="bottom.content.right" htmlTag="div" htmlId="bottom-content-right" htmlClass="col-md-4"
                           after="bottom.content.left"/>
            </container>

```

In our Magento_Catalog::product/view/details.phtml we have:
`$detailedInfoGroup = $block->getGroupSortedChildNames('detailed_info', 'getChildHtml')`
If we do not specify an argument sort_order the method returns only the last item in group.

```php
public function getGroupSortedChildNames(string $groupName, string $callback): array
{
        $groupChildNames = $this->getGroupChildNames($groupName, $callback);
        $layout = $this->getLayout();

        $childNamesSortOrder = [];

        foreach ($groupChildNames as $childName) {
            $alias = $layout->getElementAlias($childName);
            $sortOrder = (int)$this->getChildData($alias, 'sort_order') ?? 0;

            $childNamesSortOrder[$sortOrder] = $childName;
        }

        ksort($childNamesSortOrder, SORT_NUMERIC);

        return $childNamesSortOrder;
    }
```
Or even if you specify the same sort order. 
### Proposed solution

We should put the not ordered block at the end.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @dmytro-ch 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-19938` were created

**Issue Available**: @dmytro-ch, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by elvinristi** [comment]
seems https://github.com/magento/magento2/pull/24457 merged to 2.3-develop 7days ago.
Hopefully included in `2.3.3`
Until then you can hotfix it > https://github.com/magento/magento2/pull/24457.diff

**Comment by Karlasa** [comment]
Issue was fixed in #24457

Will close this issue because of that. 

Cheers,
